pkg/client/block/client_test.go

```diff
@@ -58,60 +63,64 @@ func TestBlockClient(t *testing.T) {
-	var (
-		actualBlockCh = make(chan client.Block, 1)
-		done          = make(chan struct{}, 1)
-	)
-	go func() {
-		// Test LatestBlock method.
-		actualBlock := blockClient.LatestBlock(ctx)
-		require.Equal(t, expectedHeight, actualBlock.Height())
-		require.Equal(t, expectedHash, actualBlock.Hash())
-
-		// Test CommittedBlockSequence method.
-		blockObservable := blockClient.CommittedBlocksSequence(ctx)
-		require.NotNil(t, blockObservable)
-
-		// Ensure that the observable is replayable via Last.
-		actualBlockCh <- blockObservable.Last(ctx, 1)[0]
-
-		// Ensure that the observable is replayable via Subscribe.
-		blockObserver := blockObservable.Subscribe(ctx)
-		for block := range blockObserver.Ch() {
-			actualBlockCh <- block
-			break
-		}
-
-		// Signal test completion
-		done <- struct{}{}
-	}()
-
-	// blockAssertionLoop ensures that the blocks retrieved from both LatestBlock
-	// method and CommittedBlocksSequence method match the expected block height
-	// and hash. This loop waits for blocks to be sent on the actualBlockCh channel
-	// by the methods being tested. Once the methods are done, they send a signal on
-	// the "done" channel. If the blockAssertionLoop doesn't receive any block or
-	// the done signal within a specific timeout, it assumes something has gone wrong
-	// and fails the test.
-blockAssertionLoop:
-	for {
-		select {
-		case actualBlock := <-actualBlockCh:
-			require.Equal(t, expectedHeight, actualBlock.Height())
-			require.Equal(t, expectedHash, actualBlock.Hash())
-		case <-done:
-			break blockAssertionLoop
-		case <-time.After(blockAssertionLoopTimeout):
-			t.Fatal("timed out waiting for block event")
-		}
+	// Wait a tick for the observables to be set up. This isn't strictly
+	// necessary but is done to mitigate flakiness.
+	time.Sleep(10 * time.Millisecond)
+
+	tests := []struct {
+		name string
+		fn   func() client.Block
+	}{
+		{
+			name: "LatestBlock",
+			fn: func() client.Block {
+				return blockClient.LatestBlock(ctx)
+			},
+		},
+		{
+			name: "CommittedBlocksSequence",
+			fn: func() client.Block {
+				blockObservable := blockClient.CommittedBlocksSequence(ctx)
+				require.NotNil(t, blockObservable)
+
+				// Ensure that the observable is replayable via Last.
+				lastBlock := blockObservable.Last(ctx, 1)[0]
+				require.Equal(t, expectedHeight, lastBlock.Height())
+				require.Equal(t, expectedHash, lastBlock.Hash())
+
+				// Ensure that the observable is replayable via Subscribe.
+				blockObserver := blockObservable.Subscribe(ctx)
+				for _ = range blockObserver.Ch() {
+					// TODO_THIS_COMMIT: should we assert that this matches lastBlock?
+					break
+				}
+
+				return lastBlock
+			},
+		},
 	}
 
-	// Wait a tick for the observables to be set up.
-	time.Sleep(time.Millisecond)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var eitherActualBlockCh = make(chan client.Block, 10)
+
+			// Run test functions concurrently because they can block, leading
```

Comment:
The words `functions concurrently` make it sound like the test functions run concurrently.

```suggestion
			// Run test function asynchronously because they can block, leading
```